servers/src/main/java/tachyon/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -41,6 +42,7 @@
 /**
  * This class is a Thrift handler for file system master RPCs invoked by a Tachyon client.
  */
+@ThreadSafe
```

Comment:
agree. There are public methods not synchronized in `FileSystemMaster` and `BlockMaster`.  Let us double check to make sure `FileSystemMaster` and `BlockMaster` thread-safe first, before marking those service handler thread-safe.